### PR TITLE
feat(organizations): add organization creation page

### LINKED ISSUE #7738: Feature: Add organization creation page
### Is your feature request related to a problem? Please describe

As part of the Enterprise Organizations feature, users need a way to create a new organization through the UI. The `new`/`create` controller actions exist, but there's no frontend page to actually create an organization.

### Describe the solution you'd like

Add a `/organizations/new` page with a form to create an organization. The form should collect name, description, location, visibility (public/private), external links, and a logo, with client-side interactivity (character counters, live provider-logo detection for links, drag-and-drop logo upload).

On submit, the organization is created and the creator is automatically added as an admin, then redirected to the new organization's overview page.

Introduces a reusable `OrganizationFormComponent` that will later be shared with the settings (edit) page.

### Describe alternatives you've considered

N/A

### Additional context

GSoC 26 - Project 6 Enterprise & Institutional Organization Features

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by naman79820**
Fixes #7738 

 

#### Describe the changes you have made in this PR -
Adds an organization creation page at /organizations/new. The form collects name, description, location, visibility (public/private), external links, and a logo. On submit, the organization is created and the creator is automatically added as an admin, then redirected to the new organization's overview.

Introduces a reusable OrganizationFormComponent (shared with the future settings/edit page) with client-side interactivity: character counters, live provider-logo detection for links, and drag-and-drop logo upload.

### Screenshots of the UI changes (If any) -

https://github.com/user-attachments/assets/c64c2e66-0375-4a09-8d8c-6243021e3609

---

## Code Understanding and AI Usage

**Did you use AI generated code (ChatGPT, Claude, Copilot, etc.) in any part of this PR? **
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
The new and create controller actions already existed, but there was no page to actually create an organization, so I added the form.

I built the form as a reusable component rather than inline so the settings/edit page can reuse it later. form_with(model:) already handles POST vs PATCH depending on whether the org is new or existing, and I added a small cancel_path helper so Cancel goes to the org overview when editing or the orgs list when creating.

Creation happens inside a transaction that saves the org and adds the creator as an admin together, so an org is never left without an admin. The Stimulus controller handles the live stuff in the form  the character counters, the link icons that update as you type, and the drag-and-drop logo upload. For the link icons I match the domain with an endsWith check so real subdomains work but something like github.com.evil.com doesn't get treated as GitHub.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [ ] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## New Features
- Added organization creation and settings forms for details, visibility, external links, and logos.
- Added drag-and-drop logo uploads with previews, removal controls, accessible interactions, and filename feedback.
- Added dynamic social-link fields with provider-specific icons and support for up to five links.
- Added validation feedback, character counters, localized guidance, and creation-page content.

## Style
- Added responsive form styling, upload-zone states, icons, text wrapping, and fixed-size logo previews.

## Bug Fixes
- Reduced the maximum organization location length to 50 characters.
- Added logo validation for supported image formats and a 2 MB file-size limit.

### CODE REVIEW

File: app/component/organization_form_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  <%# Location %>
+  <div class="mb-3">
+    <label class="form-label fw-semibold" for="organization_location"><%= t("organizations.form.location_label") %></label>
+    <div class="form-text mt-0 mb-2"><%= t("organizations.form.location_hint") %></div>
+    <div class="input-group">
+      <span class="input-group-text"><i class="fa fa-map-marker-alt org-form-icon" aria-hidden="true"></i></span>
+      <%= form.text_field :location, id: "organization_location", class: "form-control",
+            placeholder: t("organizations.form.location_placeholder"), maxlength: 30 %>
+    </div>
+    <div class="text-end"><small id="org-location-counter" class="form-text"></small></div>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/component/organization_form_component.html.erb:53]: Personally i think max length should be 30 words, even X has 30 words max length

File: app/javascript/controllers/organization_form_controller.js [hunk 2]
```diff
@@ -0,0 +1,143 @@
+/* eslint-disable class-methods-use-this */
+import { Controller } from 'stimulus';
+
+const MAX_LINKS = 5;
+
+export default class extends Controller {
+    connect() {
+        this.setupCounters();
+        this.setupLinks();
+        this.setupLogo();
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/javascript/controllers/organization_form_controller.js:11]: what should be the approach for this fix? @coderabbitai

File: app/component/organization_form_component.html.erb (hunk 1 above)
[the remark this answers is not in the document]
Comment (naman79820) [reply app/component/organization_form_component.html.erb:53]: ok i have updated it to 50 actually

File: app/models/organization.rb [hunk 3]
```diff
@@ -55,4 +56,13 @@ def links_count_within_limit
...
       errors.add(:links, "cannot have more than #{MAX_LINKS} links") if links.size > MAX_LINKS
     end
+
+    def logo_must_be_valid_image
+      return unless logo.attached?
+
+      acceptable_types = ["image/png", "image/jpeg", "image/svg+xml"]
+      errors.add(:logo, "must be a PNG, JPEG, or SVG image") unless acceptable_types.include?(logo.content_type)
+
+      errors.add(:logo, "must be smaller than 2 MB") if logo.byte_size > 2.megabytes
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/models/organization.rb]: recheck i have updated it @coderabbitai

[the remark this answers is not in the document]
Comment (naman79820) [reply app/models/organization.rb]: actaully leave it, I should use i18n.

[the remark this answers is not in the document]
Comment (naman79820) [reply app/models/organization.rb]: what should be the approach for i18n? as i dont see any other models with yml file

[the remark this answers is not in the document]
Comment (naman79820) [reply app/models/organization.rb]: done

File: app/component/organization_form_component.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    attr_reader :organization
+
+    def existing_links
+      return [] unless organization.links.is_a?(Array)
+
+      organization.links.compact_blank
+    end
+
+    def cancel_path
+      if organization.persisted?
```
Comment (vedant-jain03) [inline app/component/organization_form_component.rb:20]: Let's use action names instead of checking weather organization exist. Assuming this is for edit form -> cancel

File: app/javascript/controllers/organization_form_controller.js [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+            img.src = logos.logoDefault;
+            return;
+        }
+        const is = (host, domain) => host === domain || host.endsWith(`.${domain}`);
+        if (is(hostname, 'github.com')) img.src = logos.logoGithub;
+        else if (is(hostname, 'twitter.com') || is(hostname, 'x.com')) img.src = logos.logoX;
+        else if (is(hostname, 'linkedin.com')) img.src = logos.logoLinkedin;
+        else if (is(hostname, 'facebook.com')) img.src = logos.logoFacebook;
+        else if (is(hostname, 'youtube.com') || is(hostname, 'youtu.be')) img.src = logos.logoYoutube;
+        else img.src = logos.logoDefault;
```
Comment (vedant-jain03) [inline app/javascript/controllers/organization_form_controller.js:132]: Since we already have this in BE, it would be harder to maintain since now there is 2 SSOT, we should be doing the JSON here as well, and the JSON should be sharable across both!

**Comment by vedant-jain03** [comment]
Note on UI/UX:

- Show the uploaded logo in the creation form as well and not just the name
- In the overview page, there are 2 times the title being rendered - I think we can simply remove the top one!

**Comment by vedant-jain03** [comment]
Nice work here @naman79820 , changes LGTM 🚀

**Review by vedant-jain03 — APPROVED** [review]
